extra_data/reader.py

```diff
@@ -1447,9 +1447,15 @@ def open_run(proposal, run, data='raw', include='*'):
 
     prop_dir = find_proposal(proposal)
 
-    if isinstance(run, int):
-        run = 'r{:04d}'.format(run)
-    elif not run.startswith('r'):
-        run = 'r' + run.rjust(4, '0')
+    if isinstance(run, str):
+        if run.startswith('r'):
+            run = run[1:]
+    try:
+        run = operator.index(run)
```

Comment:
`index()` doesn't accept strings, so we should put the try inside an `else` block.

And we'll need to import `operator`, as this is the first use in this module.